File: erezervisi_api/eRezervisi/eRezervisi.Common.Dtos/User/UserUpdateDto.cs
namespace eRezervisi.Common.Dtos.User
{
    public class UserUpdateDto
    {
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string Phone { get; set; } = null!;
        public string Address { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string? Image { get; set; }
        public string Username { get; set; } = null!;
        public string? NewPassword { get; set; } = null!;
    }
}

File: erezervisi_mobile/erezervisi_mobile/lib/models/requests/user/user_update_dto.dart
class UserUpdateDto {
  String firstName;
  String lastName;
  String phone;
  String address;
  String email;
  String? imageBase64;
  String? imageFileName;
  String username;
  String? newPassword;

  UserUpdateDto({
    required this.firstName,
    required this.lastName,
    required this.phone,
    required this.address,
    required this.email,
    this.imageBase64,
    this.imageFileName,
    required this.username,
    this.newPassword,
  });

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'phone': phone,
      'address': address,
      'email': email,
      'imageBase64': imageBase64,
      'imageFileName': imageFileName,
      'username': username,
      'newPassword': newPassword
    };
  }
}
